extension-manifest-v3/src/background/utils/map.js

```diff
@@ -96,6 +96,13 @@ export default class AutoSyncingMap {
     return this.inMemoryMap.get(key);
   }
 
+  async getAll() {
+    this._warnIfOutOfSync();
```

Comment:
This function is unused now, right? Maybe we should remove it?

If keeping, we should remove _warnIfOutOfSync since it is async, waits for _pending and thus is always consistent from the perspective of the caller.

Reply:
You might be right, but after switching to indexedDB, changes in `map` are rather cosmetic, allowing initialize more than one map instance (it was a singleton before). If you think there are things to clean up, please make a separate issue for this.

Reply:
From what I see, the PR here only adds the extra function "getAll" though it no longer uses it.

Thinking about it again, I would rather remove it. Mostly since the class was originally designed to provide synchronous access (to be only used in situations where data loss is acceptable). Unless there is a reason, I would keep all functions synchronous (or only expose an explicit wait function to guarantee that loading is finished).

Reply:
Good point 👍